Library/Homebrew/formula.rb

```diff
@@ -2038,11 +2057,16 @@ def needs(*standards)
     #
     # The test will fail if it returns false, or if an exception is raised.
     # Failed assertions and failed `system` commands will raise exceptions.
-
     def test(&block)
       define_method(:test, &block)
     end
 
+    # TODO
+    def pour_bottle?(reason, &block)
+      @pour_bottle_reason = reason
+      @pour_bottle_block = block
```

Comment:
Some suggestions:
* We should allow no reason. The current `pour_bottle? nil` looks weird.
* If no reason is given, it may be a good to offer a default one.
* Instead of introducing new `pour_bottle_block` variable, we can store it in `pour_bottle?` as `define_method(: pour_bottle?, &block)`. As result, `Formula#pour_bottle?` should be tagged as private instead of deprecated. And you don't need to check `pour_bottle_block` existence in `formula_installer`. In general, it should be help to simplify the code and make it more readable.